firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarAction.swift

```diff
@@ -49,19 +49,32 @@ class ToolbarMiddlewareAction: Action {
     let buttonType: ToolbarActionState.ActionType?
     let buttonTapped: UIButton?
     let gestureType: ToolbarButtonGesture?
+    let url: URL?
+    let traitCollection: UITraitCollection?
+    let canGoForward: Bool?
+    let canGoBack: Bool?
```

Comment:
Is `urlDidChange` likely the only action type that will utilize these new properties? If so would it make sense for us to further subclass `ToolbarMiddlewareAction` into a more specific subclass action (so that the other actions don't include these properties).